File: frontend/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";
//entity with some properties
export class Purchase {
    customer!: Customer ;
    shippingAddress!: Address ;
    billingAddress!: Address ;
    order!: Order;
    orderItems!: OrderItem[];
}

File: backend/src/main/java/com/tarek/ecommerceapp/dto/Purchase.java
package com.tarek.ecommerceapp.dto;


import com.tarek.ecommerceapp.entity.Address;
import com.tarek.ecommerceapp.entity.Customer;
import com.tarek.ecommerceapp.entity.Order;
import com.tarek.ecommerceapp.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase { 
    private Customer customer;
    private Order order;
    private Address shippingAddress;
    private Address billingAddress;
    private Set<OrderItem> orderItems;
}
